src/VirtualTable/BodyGrid.tsx

```diff
@@ -109,6 +148,22 @@ const Grid = React.forwardRef<GridRef, GridProps>((props, ref) => {
   };
 
   const extraRender: ListProps<any>['extraRender'] = info => {
+    // Get current scroll info
+    const currentScrollInfo = listRef.current?.getScrollInfo();
+
+    // If we're expanding/collapsing the first row, don't do any automatic scrolling
+    if (firstRowExpandChangeRef.current && currentScrollInfo) {
+      firstRowExpandChangeRef.current = false;
+
+      // Use rAF to execute after the current render cycle
+      requestAnimationFrame(() => {
+        // Keep the scroll position at its current position rather than jumping
+        if (listRef.current) {
+          listRef.current.scrollTo({ top: 0 });
```

Comment:
This seems be the `rc-virtual-list` length extends bug. Not a better place to fix in `rc-table`